src/lib/djerba/plugins/fusion/plugin.py

```diff
@@ -63,12 +67,129 @@ def sort_by_actionable_level(row):
             }
             gene_info = []
             treatment_opts = []
+
+        # Processing fusions and generating blob URLs
+        tsv_file_path = wrapper.get_my_string(fc.ARRIBA_PATH)
+        base_dir = (directory_finder(self.log_level, self.log_path).get_base_dir())
+        fusion_dir = os.path.join(base_dir, "plugins/fusion")
+        json_template_path = os.path.join(fusion_dir, fc.JSON_TO_BE_COMPRESSED)
+        output_dir = self.workspace.get_work_dir()
+        unique_fusions = list({item["fusion"] for item in results[fc.BODY]})
+        fusion_url_pairs = []
+
+        for fusion in unique_fusions:
+            try:
+                fusion, blurb_url = self.process_fusion(config, fusion, tsv_file_path, json_template_path, output_dir)
+                fusion_url_pairs.append([fusion, blurb_url])
+
+            except ValueError as e:
```

Comment:
What kind of `ValueError` do we expect here? Is it definitely OK to log the error and continue when this happens?

Reply:
ValueError can be raised:  
- If fusion doesn’t match the format gene1::gene2 
- If we don’t find a match in the arriba tsv file (and this actually happened with the old test data, the old arriba file didn’t have the BRAF::SND1 fusion)
I think it’s okay to proceed if the error is isolated to one fusion but it wouldn’t be okay if the specific fusion being skipped is essential and actionable.  
I improved the error messages and now we have a variable to log the skipped fusions, if any! Let me know if you have other ideas!

Reply:
See my comment on line 127 -- the general idea is OK, but would be much safer to use a custom error class

Reply:
Added a custom error class